Source/Libraries/FaultData/DataWriters/Emails/ConfigurationLoader.cs

```diff
@@ -114,15 +106,58 @@ private IEnumerable<string[]> LoadXDASettings(AdoDataConnection connection)
             return settingList.Select(setting => ToArray(setting.Name, setting.Value));
         }
 
-        private IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection)
+        protected virtual IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection) =>
+            Enumerable.Empty<string[]>();
+
+        protected string[] ToArray(string key, string value) =>
+            key.Split('.').Concat(new[] { value }).ToArray();
+
+        #endregion
+
+        #region [ Static ]
+
+        // Static Fields
+        private static readonly ILog Log = LogManager.GetLogger(typeof(ConfigurationLoader));
+
+        // Static Properties
+        private static ConnectionStringParser<SettingAttribute, CategoryAttribute> ConnectionStringParser { get; } =
+            new ConnectionStringParser<SettingAttribute, CategoryAttribute>();
+
+        #endregion
+    }
+
+    public class ConfigurationLoader<T> : ConfigurationLoader where T : EmailDataSourceSettingBase, new()
+    {
+        #region [ Constructors ]
+
+        public ConfigurationLoader(int dataSourceEmailTypeID, Func<AdoDataConnection> connectionFactory)
+            : base(connectionFactory)
+        {
+            DataSourceEmailTypeID = dataSourceEmailTypeID;
+        }
+
+        #endregion
+
+        #region [ Properties ]
+
+        private int DataSourceEmailTypeID { get; }
+
+        #endregion
+
+        #region [ Methods ]
+
+        protected override IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection)
         {
-            if (DataSourceEmailTypeID is null)
-                return Enumerable.Empty<string[]>();
+            TableOperations<T> settingTable = new TableOperations<T>(connection);
+            string dataSourceEmailTypeIDFieldName = settingTable
+                .GetFieldNames()
+                .FirstOrDefault(name => name.EndsWith("DataSourceEmailTypeID"));
+
+            string query = dataSourceEmailTypeIDFieldName + " = {0}";
 
-            TableOperations<TriggeredEmailDataSourceSetting> settingTable = new TableOperations<TriggeredEmailDataSourceSetting>(connection);
-            List<TriggeredEmailDataSourceSetting> settingList = settingTable.QueryRecordsWhere("TriggeredEmailDataSourceEmailTypeID = {0}", DataSourceEmailTypeID).ToList();
+            List<EmailDataSourceSettingBase> settingList = settingTable.QueryRecordsWhere(query, DataSourceEmailTypeID).ToList<EmailDataSourceSettingBase>();
```

Comment:
I think this is okay, but just as an FYI, it shouldn't be necessary to explicitly specify `EmailDataSourceSettingBase` on the call to `ToList()` because of the `T : EmailDataSourceSettingBase` constraint. Leaving the type as `List<T>` would have worked just the same.